### PR TITLE
added glob for repository lookup

### PR DESCRIPTION
**Pull request by kikass13**
see https://github.com/modm-io/lbuild/issues/70

feature:
 * optional repository glob in project.xml  

needed:
- [x] probably unit test for this

### CODE REVIEW

File: lbuild/config.py [hunk 1]
```diff
@@ -222,6 +223,13 @@ def from_file(configfile, parent=None):
             rfilename = ConfigNode._rel_path(repopath, configpath)
             config._repositories.append(rfilename)
 
+        # Load glob repositories (optional)
+        for glob_node in xmltree.iterfind("repositories/glob/path"):
+            globPath = glob_node.text.format(cache=config._cachefolder)
+            for f in glob.glob(globPath):
```
Comment (salkinium) [inline lbuild/config.py]: rglob? `ext/**/repo.lb` could be useful.

**Review by salkinium — COMMENTED** [review]
I'll add some code to make the `<path>` node optional also for `<repository>`

File: lbuild/config.py (hunk 1 above)
Comment (kikass13) [reply lbuild/config.py]: Dunno, I didn't want to go "that far" by doing it recursive as well :) should probably be configurable with `recursive=False` by default.

Comment (salkinium) [reply lbuild/config.py]: Ah, you're right, glob alone already supports `**`, and you can get the same behavior as rglob by prefixing the path with `**/`.

**Comment by salkinium** [comment]
> I'll add some code to make the `<path>` node optional also for `<repository>`

I cannot do that since XSD schema doesn't support it. Hmpf.

**Comment by salkinium** [comment]
I changed it to `<glob>pattern</glob>` only, since the `<path>` thing is complemented with `<vcs>` for `lbuild init` which is not documented and not officially supported, but still necessary for backwards compatiblity.

**Comment by kikass13** [comment]
I'm happy that I can Inroduce this feature into our workspace. This will help me organize our (I don't know how many) firmware applications and keep me more sane in the process (:

**Pull request merged**